### ISSUE TITLE
Calling receiveBatch twice leads to a hang

### ISSUE THREAD

**Issue by AlexGhiondea** [Azure/azure-service-bus-node#31]
Assuming there are at least 2 messages in the queue, doing the following leads to a hang:

```ts
const msg = await client.receiveBatch(1);
const msg = await client.receiveBatch(1);
```

This is because the batching receiver will create a new batching receiver for every call, without closing the previous one.

**Comment by amarzavery** [comment]
This is as per design. It is expected to not close the batching receiver link. 2nd batch receive call should reuse the link created from the first one if the link is already open. This is done because link creation is expensive operation. Hence we try to minimize link creation wherever possible.

If you received the messages in peeklock mode then you are expected to settle the first message before you can get the second message.

Try setting the receiveMode property to receiveAndDelete and then run the test again OR when you receive the list of first set of messages, please dispose them and then call await receiveBatch again

**Comment by ramya-rao-a** [comment]
> 2nd batch receive call should reuse the link created from the first one if the link is already open.

Agreed. But this isnt what is happening. What is happening is that we are creating a new batchingReceiver for every receiveBatch call.

> If you received the messages in peeklock mode then you are expected to settle the first set of messages before you can get the second set of messages

> Try setting the receiveMode property to receiveAndDelete and then run the test again

Even if we settle the messages using `complete()` before calling the next batch or if we change to receiveAndDelete, it doesnt change the hanging issue which is due to the fact that earlier batchingRecievers never get closed. Only the last one gets closed.

Checkout #32

**Comment by ramya-rao-a** [comment]
Fixed with #32
